io-ballerina/src/io/bytes/channel_byte_io.bal

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.'value;
+
+# Read the entire channel content as a byte array.
+# ```ballerina
+# byte[]|io:Error content = io:channelReadBytes(readableChannel);
+# ```
+# + readableChannel - A readable channel
+# + return - Either a readonly byte array or `io:Error`
+public function channelReadBytes(ReadableChannel readableChannel) returns @tainted readonly & byte[]|Error {
+    if (readableChannel is ReadableByteChannel) {
+        var result = readableChannel.readAll();
+        var closeResult = readableChannel.close();
+        return result;
+    } else {
+        TypeMismatchError e = TypeMismatchError("Expected ReadableByteChannel but found a " + 'value:toString(typeof 
+        readableChannel));
+        return e;
```

Comment:
```suggestion
        return TypeMismatchError("Expected ReadableByteChannel but found a " + 'value:toString(typeof 
        readableChannel));
```

Reply:
Fixed all in https://github.com/ballerina-platform/module-ballerina-io/pull/31/commits/9b31402ee738e6603757d75a57a7a745f06549fa